src/widgets/Notebook.cpp

```diff
@@ -325,6 +325,55 @@ void Notebook::setAllowUserTabManagement(bool value)
     this->allowUserTabManagement_ = value;
 }
 
+bool Notebook::getShowTabs() const
+{
+    return this->showTabs_;
+}
+
+void Notebook::setShowTabs(bool value)
+{
+    this->showTabs_ = value;
+
+    this->performLayout();
+    for (auto &item : this->items_)
+    {
+        item.tab->setHidden(!value);
+    }
+
+    this->setShowAddButton(value);
+
+    // show a popup upon hiding tabs
+    if (!value && getSettings()->informOnTabVisibilityToggle.getValue())
+    {
+        QMessageBox msgBox;
+        msgBox.setText("Hide tabs");
+        msgBox.setInformativeText(
+            "You've just hidden your tabs.\nYou can (un)hide those with "
+            "either keyboard shortcut (Default: Ctrl+U) or select an option "
+            "in split header's menu.");
+        msgBox.addButton(QMessageBox::Ok);
+        msgBox.addButton("Don't show again", QMessageBox::YesRole);
+
+        msgBox.setDefaultButton(QMessageBox::Ok);
+
+        auto picked = msgBox.exec();
+
+        switch (picked)
+        {
+            case QMessageBox::Ok: {
+                return;
+            }
+            break;
+
+            case QMessageBox::YesRole:
+            default: {
+                getSettings()->informOnTabVisibilityToggle.setValue(false);
+            }
+            break;
+        }
+    }
```

Comment:
```suggestion
    if (!value && getSettings()->informOnTabVisibilityToggle.getValue())
    {
        QMessageBox msgBox;
        msgBox.setText("Hide tabs");
        msgBox.setInformativeText(
            "You've just hidden your tabs.\nYou can (un)hide those with "
            "either keyboard shortcut (Default: Ctrl+U) or select an option "
            "in split header's menu.");
        msgBox.addButton(QMessageBox::Ok);
        const auto *dsaButton =
            msgBox.addButton("Don't show again", QMessageBox::YesRole);

        msgBox.setDefaultButton(QMessageBox::Ok);

        msgBox.exec();

        if (msgBox.clickedButton() == dsaButton)
            getSettings()->informOnTabVisibilityToggle.setValue(false);
    }
```
I think this is "more correct" (or at least easier to understand) since comparing the *result* of the message box to a standard button makes little sense conceptually.

Reply:
Sure I agree, didn't know you can use `.clickedButton()` for that